gst-plugins-bad-1.18.0 does not build on x86_64-linux

**Describe the bug**
`gst-plugins-bad` version 1.18.0 fails to build.

**To Reproduce**
Steps to reproduce the behavior:
1. `nix-build --no-out-link -E 'with import <nixpkgs> { }; gst_all_1.gst-plugins-bad'`

**Expected behavior**
Build properly completes.

**Logs**
```
FAILED: ext/spandsp/libgstspandsp.so.p/gstspanplc.c.o
gcc -Iext/spandsp/libgstspandsp.so.p -Iext/spandsp -I../ext/spandsp -I. -I.. -I/nix/store/hap0645fi59ch3xzsaxwhci12lhf8mli-gst-plugins-base-1.18.0-dev/include/gstreamer-1.0 -I/nix/store/np39z34vyzjfm6w67x2yw4fpsw27s5yj-gstreamer-1.18.0-dev/include/gstreamer-1.0 -I/nix/store/k0hqxfw4r08c0gk10a1g1sdy371xjhyd-glib-2.66.2-dev/include -I/nix/store/k0hqxfw4r08c0gk10a1g1sdy371xjhyd-glib-2.66.2-dev/include/glib-2.0 -I/nix/store/jckkjnmp03266z8yfw403wnzbwkn4hpk-glib-2.66.2/lib/glib-2.0/include -I/nix/store/ary678h9za0pvznv263ff9q4xyfda9cy-spandsp-3.0.0-dev/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -fvisibility=hidden -fno-strict-aliasing -Wmissing-prototypes -Wdeclaration-after-statement -Wold-style-definition -Wmissing-declarations -Wredundant-decls -Wwrite-strings -Wformat -Wformat-security -Winit-self -Wmissing-include-dirs -Waddress -Wno-multichar -Wvla -Wpointer-arith -fPIC -DHAVE_CONFIG_H -MD -MQ ext/spandsp/libgstspandsp.so.p/gstspanplc.c.o -MF ext/spandsp/libgstspandsp.so.p/gstspanplc.c.o.d -o ext/spandsp/libgstspandsp.so.p/gstspanplc.c.o -c ../ext/spandsp/gstspanplc.c
../ext/spandsp/gstspanplc.c: In function 'gst_span_plc_create_stats':
../ext/spandsp/gstspanplc.c:87:45: error: dereferencing pointer to incomplete type 'plc_state_t' {aka 'struct plc_state_s'}
   87 |         "pitch", G_TYPE_INT, self->plc_state->pitch,
      |                                             ^~
```

Full build log: http://u.dk0.us/31119-gst.log

**Additional context**
This is blocking rebuilds.

**Notify maintainers**
@matthewbauer @jtojnar 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.0-rc2, NixOS, 21.03pre251987.2de60095d2b (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.8`
 - channels(root): `"home-manager, nixos-21.03pre252892.1491bab2b11"`
 - channels(anna): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: gst_all_1
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/libraries/spandsp/3.nix
+++ b/pkgs/development/libraries/spandsp/3.nix
@@ -1,0 +1,24 @@
+{ stdenv, fetchFromGitHub, audiofile, libtiff, autoreconfHook }:
+stdenv.mkDerivation rec {
+  version = "3.0.0";
+  pname = "spandsp";
+  src = fetchFromGitHub {
+    owner = "freeswitch";
+    repo = pname;
+    rev = "6ec23e5a7e411a22d59e5678d12c4d2942c4a4b6"; # upstream does not seem to believe in tags
+    sha256 = "03w0s99y3zibi5fnvn8lk92dggfgrr0mz5255745jfbz28b2d5y7";
+  };
+
+  outputs = [ "out" "dev" ];
+
+  nativeBuildInputs = [ autoreconfHook ];
+  propagatedBuildInputs = [ audiofile libtiff ];
+
+  meta = {
+    description = "A portable and modular SIP User-Agent with audio and video support";
+    homepage = "https://github.com/freeswitch/spandsp";
+    platforms = with stdenv.lib.platforms; linux;
+    maintainers = with stdenv.lib.maintainers; [ ajs124 ];
+    license = stdenv.lib.licenses.gpl2;
+  };
+}

--- a/pkgs/development/libraries/spandsp/default.nix
+++ b/pkgs/development/libraries/spandsp/default.nix
@@ -1,24 +1,23 @@
-{ stdenv, fetchFromGitHub, audiofile, libtiff, autoreconfHook }:
+{stdenv, fetchurl, audiofile, libtiff}:
 stdenv.mkDerivation rec {
-  version = "3.0.0";
+  version = "0.0.6";
   pname = "spandsp";
-  src = fetchFromGitHub {
-    owner = "freeswitch";
-    repo = pname;
-    rev = "6ec23e5a7e411a22d59e5678d12c4d2942c4a4b6"; # upstream does not seem to believe in tags
-    sha256 = "03w0s99y3zibi5fnvn8lk92dggfgrr0mz5255745jfbz28b2d5y7";
+  src=fetchurl {
+    url = "https://www.soft-switch.org/downloads/spandsp/spandsp-${version}.tar.gz";
+    sha256 = "0rclrkyspzk575v8fslzjpgp4y2s4x7xk3r55ycvpi4agv33l1fc";
   };
 
   outputs = [ "out" "dev" ];
 
-  nativeBuildInputs = [ autoreconfHook ];
-  propagatedBuildInputs = [ audiofile libtiff ];
-
+  propagatedBuildInputs = [audiofile libtiff];
   meta = {
     description = "A portable and modular SIP User-Agent with audio and video support";
-    homepage = "https://github.com/freeswitch/spandsp";
+    homepage = "http://www.creytiv.com/baresip.html";
     platforms = with stdenv.lib.platforms; linux;
     maintainers = with stdenv.lib.maintainers; [raskin];
     license = stdenv.lib.licenses.gpl2;
+    downloadPage = "http://www.soft-switch.org/downloads/spandsp/";
+    inherit version;
+    updateWalker = true;
   };
 }

--- a/pkgs/servers/sip/freeswitch/modules.nix
+++ b/pkgs/servers/sip/freeswitch/modules.nix
@@ -9,7 +9,7 @@
 , ffmpeg
 , libmysqlclient
 , postgresql
-, spandsp
+, spandsp3
 , sofia_sip
 }:
 
@@ -70,7 +70,7 @@ in
     snom = mk "applications/mod_snom" [];
     sonar = mk "applications/mod_sonar" [];
     soundtouch = mk "applications/mod_soundtouch" [];
-    spandsp = mk "applications/mod_spandsp" [ spandsp ];
+    spandsp = mk "applications/mod_spandsp" [ spandsp3 ];
     spy = mk "applications/mod_spy" [];
     stress = mk "applications/mod_stress" [];
     translate = mk "applications/mod_translate" [];

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -15794,6 +15794,7 @@ julia_15 = callPackage ../development/compilers/julia/1.5.nix {
   soundtouch = callPackage ../development/libraries/soundtouch {};
 
   spandsp = callPackage ../development/libraries/spandsp {};
+  spandsp3 = callPackage ../development/libraries/spandsp/3.nix {};
 
   spaceship-prompt = callPackage ../shells/zsh/spaceship-prompt {};
 

